app/models/Sort.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import models.Sort.Order
+import models.Sort.Order.{Ascending, Descending}
+import org.mongodb.scala.bson.conversions.Bson
+import org.mongodb.scala.model.Indexes.{ascending, descending}
+
+sealed trait Sort {
+  self: Order =>
+  val field: String
+  val orderBy: String
+  val convertParams: String
```

Comment:
`val convertParams: String = this.toString` then it can be removed from all of the below case objects